src/api/PDFFont.ts

```diff
@@ -7,6 +7,18 @@ import {
   StandardFontEmbedder,
 } from 'src/core';
 import { assertIs } from 'src/utils';
+import { BoundingBox } from 'src/types/fontkit';
+
+export interface FontMetrics {
+  /** The font's ascender */
+  ascent: number | void;
+  /** The font's descender */
+  descent: number | void;
+  /** The font's line gap if known; only works on custom fonts currently */
+  lineGap: number;
+  /** Font bounding box */
+  bbox: BoundingBox,
+}
```

Comment:
Please use `undefined` (or, better, an optional property `?:`) instead of `void`. Also, `lineGap` should be optional as well, since it is not present on custom fonts. Users may choose to default the value to zero if they wish, but pdf-lib should not hide the fact that it isn't always available from them.
```suggestion
export interface FontMetrics {
  /** The font's ascender */
  ascent?: number;
  /** The font's descender */
  descent?: number;
  /** The font's line gap, if it's accessible (currently only accessible on custom fonts) */
  lineGap?: number;
  /** Font bounding box */
  bbox: BoundingBox,
}
```

Reply:
I'm a bit confused; the custom fonts is where I *have* seen lineGap present, standard fonts does not seem to have them. If not specified the line gap would be 0, since there is then not a gap, so I feel that is the appropriate way to handle it.